tasks/artifact.xml

```diff
@@ -0,0 +1,318 @@
+        artifact.git.remote_branch - if not configured, uses the current branch with a prefix
+        artifact.prefix - if not configured, uses 'artifact-'
+        artifact.git.remote_base_branch - branch off of this base branch if the remote_branch does not exist
+        artifact.git.remote_name - 'origin'
+        artifact.gitignore_template - defaults to conf/artifact-gitignore within the-build
+        artifact.readme_template - defaults to conf/artifact-README.md within the-build
+
+      Run time flags:
+        push
+
+      -->
+
+
+    <!--
+      Target: build-artifact
+      -->
+    <target name="artifact-main" description="Build and deploy the application.">
+        <!-- These two properties MUST be provided. -->
+        <fail unless="artifact.directory" message="The directory where the artifact should be built must be configured in the 'artifact.directory' property." />
+        <fail unless="artifact.git.remote" message="The remote git repository must be configured in the 'artifact.git.remote' property." />
+
+        <!-- Default property value(s). -->
+        <property name="artifact.prefix" value="artifact-" />
+        <property name="artifact.git.remote_base_branch" value="master" />
+        <property name="artifact.git.remote_name" value="origin" />
+        <property name="artifact.gitignore_template" value="${phing.dir.artifact}/../conf/artifact-gitignore" />
+        <property name="artifact.readme_template" value="${phing.dir.artifact}/../conf/artifact-README.md" />
+
+        <!-- Get the current commit, branch, message, and tag so that they can be used to label the
+             resulting artifact and reset the repository after the artifact is built. -->
+        <exec command="git rev-parse HEAD" outputProperty="artifact.git.commit" checkreturn="true" />
+        <exec command="git rev-parse --abbrev-ref HEAD" outputProperty="artifact.git.branch" checkreturn="true" />
+        <exec command="git log -1 --oneline" outputProperty="artifact.git.commit_message" />
+        <exec command="git describe --tags --exact-match" outputProperty="artifact.git.tag" returnProperty="artifact.git.no_tag" />
+
+        <!-- Create a temporary branch name based on the commit for building the artifact,
+             to avoid branch conflicts. -->
+        <property name="artifact.git.temporary_branch" value="artifact-${artifact.git.commit}" />
+        <!-- Prefix the repository tag so that we're not using the exact same tag on the
+             artifact and on the repository, to avoid confusion, especially when the
+             artifact is built on a branch of the development repository. -->
+        <property name="artifact.git.artifact_tag" value="${artifact.prefix}${artifact.git.tag}" />
+        <!-- If the remote branch isn't configured, use a remote branch based on the name
+             of the current branch. -->
+        <if>
+            <not><equals arg1="${artifact.git.remote_branch}" arg2="" /></not>
```

Comment:
Oh dang, nice logic catch -- the intent was to set the `artifact.git.remote_branch` if it was empty, so that the artifact could automatically build to a remote branch that corresponds with the local branch (e.g. `artifact-PNET-123` or `artifact-fix-sidebar`).

... But as you point out, it will not do that.